### PR TITLE
Added a new Personas module.

### PR DESCRIPTION
**Pull request by rtdurga**
## Checklist before requesting a review

- [x] I updated `CHANGELOG.MD` with a description of my changes
- [x] If the change was to a module, I ran the code validation script (`scripts/validate.sh`)
- [x] If the change was to a module, I have added thorough tests
- [x] If the change was to a module, I have added/updated the module's README.md
- [x] If a module was added, I added a reference to the module to the repository's README.md
- [x] I verified that my code deploys successfully using `seedfarmer apply`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: e5728d96c968873f3c688f76f6161d47c5eda093
* Result: SUCCEEDED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=c2ffc7bc-2363-41ad-8d1f-7e288e0b1a9d%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: ab6413af4efeeee05a98527f42512d10ff1e2a2a
* Result: SUCCEEDED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=bd84aff0-783f-41fa-97ed-b7eb663ad59f%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: 97a0270c466515d93149a1b8e4726389aaa6c3d7
* Result: SUCCEEDED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=7aa3c180-3767-4e54-8f02-9272685ea194%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: a0a0f29d60e661dcbc30fd880cf7b5f3753d0036
* Result: SUCCEEDED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=3b1aaf97-dc09-4f75-a7d2-dca1158ca600%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: modules/examples/personas/personas.py [hunk 1]
```diff
@@ -0,0 +1,606 @@
...
+        scope: Construct,
+        construct_id: str,
+        s3_bucket_prefix: str,
+        env: Environment,
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(scope, construct_id)
+
+        # Define common permissions
+        sagemaker_describe_permissions = [
```
Comment (kukushking) [inline modules/examples/personas/personas.py:20]: refactoring suggestion: we can define custom policies based on these actions

**Review by kukushking — DISMISSED** [review]

File: modules/examples/personas/README.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+- `data_engineer_role_arn`: The ARN of the Data Engineer role.
+- `ml_engineer_role_arn`: The ARN of the ML Engineer role.
+- `mlops_role_arn`: The ARN of the MLOps role.
+- `it_auditor_role_arn`: The ARN of the IT Auditor role.
+- `model_auditor_role_arn`: The ARN of the Model Auditor role.
+
+## Usage
+
+To use this module, include it in your CDK application and provide the required input parameters. For example:
```
Comment (kukushking) [inline modules/examples/personas/README.md:38]: I think this needs to be updated to reflect usage with SeedFarmer ie from a manifest. See other modules for examples.

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: 75a5ad47b475c15a8fd00fbc38c86f3e188c2c5d
* Result: SUCCEEDED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=59295e31-cc8d-4161-8fee-c9d09839705f%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: 65e865c210f20e43592ef573d301d5c76c78e436
* Result: SUCCEEDED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=4b007f06-4d30-49f7-a056-de6e5bfb321d%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: b2bd6777f4e334e9aa1dbcaf96b713d54eb6f56d
* Result: SUCCEEDED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=8ce011c5-3f6c-4f04-9bbc-a6cbdd75648d%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: e6fbe69310eeb6fe24670b1009a1e83546a6d060
* Result: SUCCEEDED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=0cbf600d-1fb0-4c79-81dc-bc8b22f5e190%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by kukushking — APPROVED** [review]

File: modules/examples/personas/deployspec.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+publishGenericEnvVariables: true
+deploy:
+  phases:
+    install:
+      commands:
+      - env
+      # Install whatever additional build libraries
+      - npm install -g aws-cdk@2.126.0
```
Comment (LeonLuttenberger) [inline modules/examples/personas/deployspec.yaml:8]: Not sure how much this matters but we should keep the version consistent (`2.137.0`)

File: modules/examples/personas/personas.py [hunk 4]
```diff
@@ -0,0 +1,610 @@
...
+            "glue:GetTrigger",
+            "glue:GetCrawler",
+            "glue:GetDevEndpoint",
+        ]
+        # ML Engineer role
+        ml_engineer_role = iam.Role(
+            self,
+            "MLEngineerRole",
+            assumed_by=iam.ServicePrincipal("sagemaker.amazonaws.com"),
+            role_name="aiops-MLEngineer",
```
Comment (LeonLuttenberger) [inline modules/examples/personas/personas.py:65]: Should we keep the `project_name-deployment_name-self.module_name` prefix for these role names? I don't think hardcoding `aiops` as a prefix is a good idea. For example, this should resolve to `aiops-personas-personas-MLEngineer`.

File: modules/examples/personas/personas.py [hunk 5]
```diff
@@ -0,0 +1,610 @@
...
+                ),
+                iam.PolicyStatement(
+                    effect=iam.Effect.ALLOW,
+                    actions=[
+                        "ecr:GetAuthorizationToken",
+                        "ecr:BatchCheckLayerAvailability",
+                        "ecr:GetDownloadUrlForLayer",
+                        "ecr:BatchGetImage",
+                    ],
+                    resources=[f"arn:{Aws.PARTITION}:ecr:{Aws.REGION}:{Aws.ACCOUNT_ID}:repository/*"],
```
Comment (LeonLuttenberger) [inline modules/examples/personas/personas.py:605]: Minor nitpick: I noticed an inconsistency three different options being used to retreive the region/account all throughout this module:
1. `AWS.REGION`
2. `env.region`
3. `this.region`

I think `this.region` (along with `this.account` and `this.partition`) would be better.

File: modules/examples/personas/app.py [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+            "DataEngineerRoleArn": stack.personas.data_engineer_role.role_arn,
+            "ITLeadRoleArn": stack.personas.it_lead_role.role_arn,
+            "BusinessAnalystRoleArn": stack.personas.business_analyst_role.role_arn,
+            "MLOpsEngineerRoleArn": stack.personas.mlops_engineer_role.role_arn,
+            "ITAuditorRoleArn": stack.personas.it_auditor_role.role_arn,
+            "ModelRiskManagerRoleArn": stack.personas.model_risk_manager_role.role_arn,
+        }
+    ),
+)
+
```
Comment (LeonLuttenberger) [inline modules/examples/personas/app.py:39]: The module isn't applying tags or CDK nag. We should include the following here:
```python
aws_cdk.Aspects.of(app).add(cdk_nag.AwsSolutionsChecks(log_ignores=True))

if app_settings.parameters.tags:
    for tag_key, tag_value in app_settings.parameters.tags.items():
        aws_cdk.Tags.of(app).add(tag_key, tag_value)

aws_cdk.Tags.of(app).add("SeedFarmerDeploymentName", app_settings.settings.deployment_name)
aws_cdk.Tags.of(app).add("SeedFarmerModuleName", app_settings.settings.module_name)
aws_cdk.Tags.of(app).add("SeedFarmerProjectName", app_settings.settings.project_name)
```

File: modules/examples/personas/stack.py [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+        self,
+        scope: Construct,
+        construct_id: str,
+        bucket_name: Optional[str],
+        env: Environment,
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(scope, construct_id)
+
+        s3_bucket_prefix = bucket_name or f"{construct_id}-bucket"
```
Comment (LeonLuttenberger) [inline modules/examples/personas/stack.py:27]: Can you confirm that this (`f"{construct_id}-bucket"`) is what we want? The readme doesn't mention it anywhere.

I think we should either make the bucket mandatory, or just not include permissions with a made-up bucket name.

**Review by LeonLuttenberger — CHANGES_REQUESTED** [review]

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: 5dd501315c3b63a0ef13df256ba04cb9ca65acc9
* Result: SUCCEEDED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=5594e17f-9296-4378-849d-c4521acca123%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: 745158bd147353559a58c3e6a9169712eeb7fab4
* Result: SUCCEEDED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=80821cc0-3512-4c86-90ea-d22904fee41f%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by kukushking — APPROVED** [review]

**Pull request merged**